### PR TITLE
Branch max quantity tests

### PR DESCRIPTION
**Pull request by Choo-Xing-Yu**
Integrate max quantity into test cases.
Add new test cases to test max quantity

Fix typical items and related test cases.
Previously DUCK item had POULTRY tagged to it.

Fix bug where adding new item without max quantity tag will delete existing item max quantity tag

Update UG. 
DG had nothing to be updated yet

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ Format: `list`
 
 Edits an existing item in the Inventory book.
 
-Format: ` edit INDEX n/NAME q/QUANTITY s/SUPPLIER [t/TAG]…​`
+Format: ` edit INDEX n/NAME q/QUANTITY s/SUPPLIER [max/MAX_QUANTITY] [t/TAG]…​`
```
Comment (saad-haider-pp) [inline docs/UserGuide.md]: I think all the arguments, except INDEX, should be optional here

File: docs/UserGuide.md [hunk 2]
```diff
@@ -198,10 +198,10 @@ Notify the user if a certain stock is below threshold
 
 | Action    | Format, Examples                                                                                    |
 |-----------|-----------------------------------------------------------------------------------------------------|
-|**Add**    | `add n/NAME q/QUANTITY s/SUPPLIER [t/TAG]...​` <br> e.g., `add n/Chicken q/3 s/ShengSiong t/Poultry` |
+|**Add**    | `add n/NAME q/QUANTITY s/SUPPLIER [max/MAX_QUANTITY] [t/TAG]...​` <br> e.g., `add n/Chicken q/3 s/ShengSiong t/Poultry` |
 |**Clear**  | `clear`                                                                                             |
 |**Delete** | `delete INDEX`<br> e.g., `delete 3`                                                                 |
-|**Edit**   | ` edit INDEX n/NAME q/QUANTITY s/SUPPLIER [t/TAG]…​`<br> e.g.,`edit 1 n/Chicken q/50`                |
+|**Edit**   | `edit INDEX n/NAME q/QUANTITY s/SUPPLIER [max/MAX_QUANTITY] [t/TAG]…​`<br> e.g.,`edit 1 n/Chicken q/50`                |
```
Comment (saad-haider-pp) [inline docs/UserGuide.md]: Likewise for here

**Review by saad-haider-pp — COMMENTED** [review]
Good job on tanking the test cases!

File: docs/UserGuide.md (hunk 1 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md]: Fixed! thanks for pointing it out!

File: docs/UserGuide.md (hunk 2 above)
Comment (Choo-Xing-Yu) [reply docs/UserGuide.md]: Same! Fixed

**Review by saad-haider-pp — APPROVED** [review]
LGTM

**Pull request merged**